### DISCUSSION TITLE
Cek Multicollinearity Untuk Multinomial Logistic Regression

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#599] [Q&A] [answered]
Saya memiliki data yang ingin saya bentuk menjadi model multinomial logistic regression, dimana dependent variable memiliki lebih dari 2 kategori. Data tersebut dapat ditemukan [disini](https://www.kaggle.com/datasets/ghassenkhaled/wine-quality-data?resource=download). Model yang ingin saya bentuk adalah logistic regression yang dapat memprediksi nilai kualitas wine berdasarkan tingkat komponen-komponen yang ada dalam wine. Kategori penilaian mulai dari 3 hingga 9, semakin tinggi semakin baik.

Saya saat ini sudah dalam pembentukan model, dan berada pada proses pengecekan apakah data saya memiliki isu mulicollinearity atau tidak. Namun ada baiknya saya menjelaskan dulu bagaimana saya membuat model saya. Dari apa yang saya mengerti, fungsi glm() , dari apa yang saya pahami, hanya dapat membentuk model logistic regression binomial, yang dimana hanya terdapat dua kategori dalam dependent variable.

Sedangkan data saya memiliki lebih dari 2 kategori, sehingga saya menggunakan kode sebagai berikut:

```
# package yang saya temukan untuk multinomial regression
library("nnet")

# kode
wine_md_all <- multinom(formula= quality ~ .,
                   data=wine_train)

wine_md_non <- multinom(formula= quality ~ 1,
                   data=wine_train)

#stepwise regression

wine_bck <- step(object = wine_md_all,
                 direction = "backward",
                 scope = list(upper=wine_md_all,
                              lower=wine_md_non),
                 trace = F)
```

Isu saya muncul ketika saya mengecek multicollinearity menggunakan fungsi vif()
```
vif(wine_bck) # wine_bck merupakan hasil backward stepwise regression
Warning: No intercept: vifs may not be sensible.     
fix_acid      vol_acid   citric_acid     res_sugar     chlorides      free_so2     total_so2 
 6.349622e+09 -9.168039e+09 -5.924124e+11  4.194655e+05 -7.587937e+14  2.540215e+03 -1.003912e+02 
      density            pH     sulphates       alcohol         color 
 4.383079e+15 -9.260988e+12 -4.586511e+12 -1.123977e+09 -2.607408e+11 
```

Saya juga ada menemukan sebuah [saran](https://stackoverflow.com/questions/67077015/vif-no-intercept-vifs-may-not-be-sensible), namun saya tidak mengerti cara intepretasinya, dan orang memberikan saran tersebut juga tidak terlalu menjelaskan hasil output
```
labels = rownames(coefficients(wine_bck)) 
ref = setdiff(wine_bck$lab,labels)
t(sapply(labels,function(i){
       dat = wine_train
       dat$quality = as.numeric(dat$quality == i)
       vif(glm(quality ~ .,data=dat,family="binomial"))
}))

Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
    fix_acid vol_acid citric_acid  res_sugar chlorides free_so2 total_so2   density        pH sulphates
4   4.505450 2.647380    2.176667   6.001777  1.807585 2.091553  3.780802  18.59503  2.796850  1.655448
5   4.589808 2.503116    1.850610   9.226485  1.596900 2.208909  4.446764  18.65808  2.845812  1.692361
6   4.073325 2.161850    1.705111   8.394183  1.787300 2.345828  4.068246  24.19363  2.434523  1.735895
7   4.193198 1.784487    1.649729   8.146394  1.866436 2.322581  3.723282  27.50923  2.147359  1.573540
8   3.414163 1.440771    1.441384   8.739265  1.802355 2.746269  3.752163  27.76346  2.037680  1.603451
9 296.515633 2.547879    2.912599 200.231082  3.598691 3.189956  2.880226 922.65373 78.315934  2.124659
    alcohol    color
4  5.572818 7.054796
5  5.245011 7.321933
6  7.848576 6.393668
7  8.431053 6.000644
8  9.049401 4.245922
9 80.035114 1.000007
```

VIF menunjukkan bahwa sebuah variable memiliki multicollinearity ketika nilai VIF diatas 10, namun bisa dilihat diatas bahwa ada hasil yang sampai 922. Jadi pertanyaan saya apakah berarti model tidak bisa digunakan karena adanya multicollinearity? Atau apakah cara interpretasinya beda dari fungsi vif() yang biasa kita gunakan?

**Comment by Yusraf002** [comment]
Dear @AGelfP 
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by divakartika** [comment, marked as answer]
Dear @AGelfP,

Dari hasil tersebut terlihat bahwa VIF > 10 pada variabel fix_acid, res_sugar, density, pH, dan alcohol. Dapat dicoba untuk exclude salah satu variabel ini secara bergantian dan cek kembali apakah masih terdapat nilai VIF > 10. Jika masih, maka perlu dilanjutkan dengan exclude variabel lainnya yang masih memiliki VIF > 10.

Best regards,
Team Algoritma

**Reply by AGelfP** [reply]
Halo Kak, maaf respons telatnya. Tadi cara ini juga sudah disarankan oleh Kak Jafar pada sesi mentoring hari ini. Hasilnya berhasil Kak. Terima kasih yaa
